File: C/bogosort/BlackBox/BlackBox.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function to check if the array is sorted
int isSorted(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        if (arr[i] < arr[i - 1]) {
            return 0; // Not sorted
        }
    }
    return 1; // Sorted
}

// Function to shuffle the array
void shuffle(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        int j = rand() % n;
        // Swap arr[i] and arr[j]
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

// Bogosort function
void bogosort(int arr[], int n) {
    srand(time(NULL));
    while (!isSorted(arr, n)) {
        shuffle(arr, n);
    }
}

int main() {
    const int SIZE = 10;
    int arr[SIZE];

    // Fill array with random values
    for (int i = 0; i < SIZE; i++) {
        arr[i] = rand() % 1000; // Random values between 0 and 9999
    }

    // Sort the array using bogosort
    bogosort(arr, SIZE);

    return 0;
}

File: Java/bogosort/BlackBox/BlackBox.java
import java.util.Random;

public class BlackBox {

    // Method to check if the array is sorted
    private static boolean isSorted(int[] array) {
        for (int i = 1; i < array.length; i++) {
            if (array[i] < array[i - 1]) {
                return false;
            }
        }
        return true;
    }

    // Method to shuffle the array
    private static void shuffle(int[] array) {
        Random rand = new Random();
        for (int i = array.length - 1; i > 0; i--) {
            int j = rand.nextInt(i + 1);
            // Swap array[i] with the element at random index
            int temp = array[i];
            array[i] = array[j];
            array[j] = temp;
        }
    }

    // Bogosort implementation
    public static void bogosort(int[] array) {
        while (!isSorted(array)) {
            shuffle(array);
        }
    }

    // Main method to test the Bogosort algorithm
    public static void main(String[] args) {
        int[] array = new int[10];
        Random rand = new Random();

        // Fill the array with random integers
        for (int i = 0; i < array.length; i++) {
            array[i] = rand.nextInt(1000); // Random integers from 0 to 9999
        }

        // Sort the array using Bogosort
        bogosort(array);
    }
}